src/NuGet.Clients/NuGet.CommandLine/MsBuildToolset.cs

```diff
@@ -0,0 +1,173 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Runtime.InteropServices.ComTypes;
+using Microsoft.VisualStudio.Setup.Configuration;
+
+namespace NuGet.CommandLine
+{
+    /// <summary>
+    /// Extended type to include Toolset data from the classic installation story and the appropriate values from
+    /// ISetupInstance in the new (with MSBuild v15.1) installation story. Allows all discovered installations
+    /// to be contained in the same collection.
+    /// </summary>
+    public class MsBuildToolset: IComparable
```

Comment:
I would expect this to be ``IComparable<MsBuildToolset>`` so that it can only be compared to the same type instead of objects.